src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+import java.util.LinkedList;
+
+public class Parser {
+    
+    public static LinkedList<String> parseAndExecute(String input, TaskList tasks, FileHandler fileHandler) 

```

Comment:
Personally, I think Parser only processes an unformatted String, i.e. a user input that must be broken down, and return one single type of object (in my case, a Command, because a user input **is** a command). With that, the Parser knows very little about other classes like TaskList and FileHandler, thus decoupling it from the rest. This is one of the software engineering principle which will help when you try to test Parser. This is so because you would only need to ensure the command is correct before you can test it properly. In the current state when you want to test Parser, you will have to make sure that TaskList and FileHandler are running correctly, which complicates the work. (Might not be depending on how you order the testings) But generally decoupling is encouraged. Hope this helps! You can also share your view on this so that we could discuss together.